src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2015 Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "simulation_defines.hpp"
+#include "particles/traits/GetAtomicNumbers.hpp"
+#include "traits/attribute/GetChargeState.hpp"
+#include "algorithms/math/floatMath/floatingPoint.tpp"
+#include "plugins/radiation/utilities.hpp"
```

Comment:
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

We are talking about two mini functors here, aren't we? ;) pls keep the includes clean, I can show you a handful of projects that ignored that and did not succeed...
-----BEGIN PGP SIGNATURE-----
Version: APG v1.1.1

iQJCBAEBCgAsBQJVkWEvJRxBeGVsIEh1ZWJsIDxheGVsLmh1ZWJsQHBsYXNtYS5u
aW5qYT4ACgkQRy0sNk4h2UvJJhAAlpl9gRPVIrz/U2LGOOupdDt7Yj/0IM5HBJRG
aeeegzKvtTvdRLqqq+/+oE1mzeB8vJYliQVVa1NG6HNfovEUo5Tcn8UVWbxtNxJY
QKCvO82CGnWZSVFk02n8Cov0pVKmGsn6JlyNeUkcMCB0JQV0O106PxZiVxvVnLVv
YMQnbI5tyr8bnwjEJ77F05vZgHQqazs3dL3+DuY9SK6OTPEheNzLDBd19LMwH8Z+
UFpMXtfdGXaAsbgjZ264CZitk/hPo5T/mkysPvmW3QfBsjw2x/GEgXU5nHrg3qsu
gY9dzUSAlCt/f0C7BIc6vWBCVUz+bQcIolM5FHJD9SlU57uJIR0N28iuw0Ilv937
2wUMI2/+z7cZEVD3AqG2+AAtxFg9bb3Zg4zJOh3x2k2XC1aqxXrt4yveRreXuNLS
BBKnnw8BlQLvX6jjLfPyYZoBb9kPJZKJa5NjOCq9eHM8hy7mmFSOHN5NZfpR2YQk
bb5fXZddn5gLy0ISX+IMtWiZUtXr5ws6ri7H4g7BJLH7/bXAbpsvYzOWRDbLlpC2
Pa6qaSejXmxn0a05qD5mFseo3DOghWu4cgg5QdPNJntGZV/tYZxCDKTGsIrJRaSt
YCmwUEH7FuWzI7R6JOZmGsiApce5MuYtbOlMnBEjkm9MNRknGyVPiTnAmL0F2yiZ
DV6d81g=
=EbtW
-----END PGP SIGNATURE-----